monkey/monkey_island/cc/services/config.py

```diff
@@ -281,3 +321,66 @@ def decrypt_ssh_key_pair(pair, encrypt=False):
             pair['public_key'] = encryptor.dec(pair['public_key'])
             pair['private_key'] = encryptor.dec(pair['private_key'])
         return pair
+
+    @staticmethod
+    def add_PBA_files(post_breach_files):
+        """
+        Interceptor of config saving process that uploads PBA files to server
+        and saves filenames and sizes in config instead of full file.
+        :param post_breach_files: Data URL encoded files
+        """
+        # If any file is uploaded
+        if any(file_name in post_breach_files for file_name in ['linux_file', 'windows_file']):
+            # Create directory for file uploads if not present
+            if not os.path.exists(UPLOADS_DIR):
+                os.makedirs(UPLOADS_DIR)
+        if 'linux_file' in post_breach_files:
+            linux_name, linux_size = ConfigService.upload_file(post_breach_files['linux_file'], UPLOADS_DIR)
+            post_breach_files['linux_file_info']['name'] = linux_name
+            post_breach_files['linux_file_info']['size'] = linux_size
+        if 'windows_file' in post_breach_files:
+            windows_name, windows_size = ConfigService.upload_file(post_breach_files['windows_file'], UPLOADS_DIR)
+            post_breach_files['windows_file_info']['name'] = windows_name
+            post_breach_files['windows_file_info']['size'] = windows_size
+
+    @staticmethod
+    def remove_PBA_files():
+        if ConfigService.get_config():
+            linux_file_name = ConfigService.get_config_value(
+                ['monkey', 'behaviour', 'custom_post_breach', 'linux_file_info', 'name'])
+            windows_file_name = ConfigService.get_config_value(
+                ['monkey', 'behaviour', 'custom_post_breach', 'windows_file_info', 'name'])
+            if linux_file_name:
+                ConfigService.remove_file(linux_file_name)
+            if windows_file_name:
+                ConfigService.remove_file(windows_file_name)
+
+    @staticmethod
+    def remove_file(file_name):
+        file_path = os.path.join(UPLOADS_DIR, file_name)
+        try:
+            if os.path.exists(file_path):
+                os.remove(file_path)
+        except OSError as e:
+            logger.error("Can't remove previously uploaded post breach files: %s" % e)
+
+
+    @staticmethod
+    def upload_file(file_data, directory):
```

Comment:
this function doesn't belong to this file. Create a new python file and move all PBA related logic from this file to the new one